packages/cln_plugin/lib/src/rpc_method/types/rpc_types.dart

```diff
@@ -0,0 +1,67 @@
+import 'package:cln_plugin/src/cln_plugin_base.dart';
+import 'package:cln_plugin/src/rpc_method/rpc_command.dart';
+
+class RPCMethod extends RPCCommand {
+  /// Method name
+  String name;
+
+  /// Usage of this method
+  String usage;
+
+  /// Description of this method
+  String description;
+
+  /// The long description of this method
+  late String longDescription;
+
+  /// Mark if the method is deprecated or not
+  bool deprecated;
+
+  RPCMethod(
+      {required this.name,
+      required this.usage,
+      required this.description,
+      required Future<Map<String, Object>> Function(Plugin, Map<String, Object>)
+          callback,
+      this.deprecated = false,
+      String? longDescription})
+      : super(callback: callback) {
+    // means if is null set the longDescription = description
+    this.longDescription = longDescription ?? description;
+  }
+
+  Map<String, dynamic> toMap() {
+    return {
+      "name": name,
+      "usage": usage,
+      "description": description,
+      "long_description": longDescription,
+      "deprecated": deprecated,
+    };
+  }
+}
+
+class RPCNotification extends RPCCommand {
+  /// Method name
+  String method;
+
+  RPCNotification(
+      {required this.method,
+      required Future<Map<String, Object>> Function(Plugin, Map<String, Object>)
+          callback})
+      : super(callback: callback);
+}
+
+class RPCHook extends RPCCommand {
+  String name;
+  List<String>? before;
+  List<String>? after;
```

Comment:
Please verify this one. :)

Reply:
This can be left empty so, you can do

```dart 
List<String> before;
List<String> after;

RPCHook({this.before = [], this.after = []})
```

Maybe the syntax is not super correct!